--- xv6-pristine/defs.h
+++ xv6-assignment5/defs.h
@@ -185,6 +185,15 @@
 void            switchkvm(void);
 int             copyout(pde_t*, uint, void*, uint);
 void            clearpteu(pde_t *pgdir, char *uva);
+int             map_shared_page(pde_t *pgdir, uint pa);
+
+// sync.c
+void            syncinit(void);
+int             shm_attach(void);
+int             seminit(int, int);
+int             semwait(int);
+int             sempost(int);
+int             semdestroy(int);
 
 // number of elements in fixed-size array
 #define NELEM(x) (sizeof(x)/sizeof((x)[0]))
